src/Components/OATHeader/OATHeader.tsx

```diff
@@ -316,34 +313,55 @@ const OATHeader = ({ dispatch, state }: OATHeaderProps) => {
     };
 
     const handleFileListChanged = async (files: Array<File>) => {
-        const items = [];
+        const newModels = [];
         if (files.length > 0) {
             const filesErrors = [];
             let modelsMetadataReference = null;
             for (const current of files) {
                 const content = await current.text();
-                const error = await parseModel(content);
-                if (!error) {
+                const JSONContent = JSON.parse(content);
```

Comment:
I think `jsonContent` would be the right capitalization here.

Reply:
Also, what happens if the file isn't valid JSON? Do we catch it somewhere?

Reply:
We are checking if `type === 'application/json'` on the previous method `onFilesUpload`

Reply:
Hmmm. Is that driven by a `.json` extension? Or does that do a deep inspection of the file?

Reply:
True, I added a validation